model/dce-stdio.cc

```diff
@@ -460,6 +412,19 @@ int dce_fileno (FILE *stream)
   NS_LOG_FUNCTION (Current () << UtilsGetNodeId () << stream);
   NS_ASSERT (Current () != 0);
   Thread *current = Current ();
+
+  // UGLY: FIXME: Use current->process->openStreams instead
+  if (current->process->pstdin && stream == *current->process->pstdin) {
+      return 0;
+  } else if (current->process->pstdout && stream == *current->process->pstdout) {
+      return 1;
+  } else if (current->process->pstderr && stream == *current->process->pstderr) {
+      return 2;
+  }
```

Comment:
I also didn't need to do this in my branch, yet everything worked fine (after using dce_fopen() in dce_freopen() and also updating offset changes in the cookie).
Sorry, can you please help me understand why we're having these explicit definitions of what is be returned for these special cases?